src/Interfaces.jl

```diff
@@ -94,22 +94,57 @@ function methodparts(IT, x::Expr)
     return methodname, args
 end
 
-requiredmethod(IT, nm, args, shouldthrow) = :(Interfaces.implemented($nm, $args, mods) || ($shouldthrow && Interfaces.missingmethod($IT, $nm, $args, mods)))
+"""
+Construct the method signature from a function name and argument types.
+
+    methodsig(:foo, Tuple{Int64, Float64, Any}) == :(foo(::Int64, ::Float64, ::Any))
+"""
+methodsig(f, args) = Expr(:call, f, unconvertargs(args)...)
+
+function requiredmethod(IT, nm, args, shouldthrow)
+    :(Interfaces.implemented($nm, $args, mods) || ($shouldthrow && Interfaces.missingmethod(debuglvl, $IT, $nm, $args, mods)))
+end
 
-function requiredreturn(T, nm, args, shouldthrow, RT_sym, __RT__)
+function requiredreturn(IT, nm, args, shouldthrow, RT_sym, __RT__)
     return quote
-        check = $(requiredmethod(T, nm, args, shouldthrow))
+        $(requiredmethod(IT, nm, args, shouldthrow))
         $RT_sym = Interfaces.returntype($nm, $args)
         # @show $RT_sym, $nm, $args, Interfaces.isinterfacetype($__RT__)
-        check |= Interfaces.isinterfacetype($__RT__) ?  Interfaces.implements($RT_sym, $__RT__) : $RT_sym <: $__RT__
-        check || ($shouldthrow && Interfaces.invalidreturntype($nm, $args, $RT_sym, $__RT__))
+        check = Interfaces.isinterfacetype($__RT__) ?  Interfaces.implements($RT_sym, $__RT__) : $RT_sym <: $__RT__
+        check || ($shouldthrow && Interfaces.invalidreturntype(debuglvl, $IT, $nm, $args, $RT_sym, $__RT__))
```

Comment:
`shouldthrow` is actually "should `throw` or log a message is the check failed", and is here so we can set it to `false` when we have `||` expressions (i.e. so we can report/throw one message for the whole `||` expression, not reporting/throwing a message for every clause in the `||` expression)

So we might want to rename that something if it's too much of a confusion/clash with `debug`/`debuglvl` ...but personally no alternative name stood out, we probably just need a comment/docstring on `toimplements!` about this